modin/experimental/core/execution/native/implementations/hdk_on_native/dataframe/dataframe.py

```diff
@@ -2505,7 +2718,12 @@ def from_arrow(cls, at, index_cols=None, index=None, columns=None):
 
         if columns is not None:
             new_columns = columns
-            new_index = pd.RangeIndex(at.num_rows) if index is None else index
+            if index is not None:
+                new_index = index
+            elif index_cols is not None:
+                new_index = None
+            else:
+                new_index = pd.RangeIndex(at.num_rows)
```

Comment:
Why not materialize the index on access as usual? This materialization might be useless if the index is not used later.